services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -519,36 +526,49 @@ async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
         # NOTE: builtin `DockerServices.update` function is very limited.
         # Using the same pattern but updating labels
 
-        try:
-            # fetch information from service name
-            service_inspect = await client.services.inspect(scheduler_data.service_name)
-            service_version = service_inspect["Version"]["Index"]
-            service_id = service_inspect["ID"]
-            spec = service_inspect["Spec"]
-
-            # compose_spec needs to be json encoded
-            # before encoding it to json and storing it
-            # in the label
-            scheduler_data_copy = deepcopy(scheduler_data)
-            scheduler_data_copy.compose_spec = json.dumps(
-                scheduler_data_copy.compose_spec
-            )
-            label_data = scheduler_data_copy.json()
-            spec["Labels"][DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL] = label_data
-
-            await client._query_json(  # pylint: disable=protected-access
-                f"services/{service_id}/update",
-                method="POST",
-                data=json.dumps(clean_map(spec)),
-                params={"version": service_version},
-            )
-        except aiodocker.exceptions.DockerError as e:
-            if not (
-                e.status == 404
-                and e.message == f"service {scheduler_data.service_name} not found"
-            ):
-                raise e
-            log.debug(
-                "Skip update for service '%s' which could not be found",
-                scheduler_data.service_name,
-            )
+        # The docker service update API is async, so `update out of sequence` error
+        # might get raised. This is caused by the `service_version` being out of sync
+        # with what is currently stored in the docker daemon.
+        async for attempt in AsyncRetrying(
+            stop=stop_after_attempt(4),
+            wait=wait_exponential(),
```

Comment:
what is the max time in secs this takes in the worse case? 4 times exponential  means ... ?

Reply:
This is the equivalent of waiting between retries [1, 4, 8, 16] seconds and than giving up